### PR TITLE
Set international postage and international flag for international letters

### PR DESCRIPTION
**Pull request by servingUpAces**
Make sure to set the postage and international flag correctly for all letters. 

There are many ways to send a letter with Notify:

1. send one-off templated letter
2. upload CSV for a templated letter
3. send template letter via the API
4. send PDF letter via the API
5. upload PDF via the UI

Possible postage types are first | second | europe | rest-of-world.

### CODE REVIEW

File: app/notifications/process_notifications.py [hunk 1]
```diff
@@ -147,7 +146,8 @@ def persist_notification(
     elif notification_type == EMAIL_TYPE:
         notification.normalised_to = format_email_address(notification.to)
     elif notification_type == LETTER_TYPE:
-        notification.postage = postage or template_postage
+        notification.postage = postage
+        notification.international = True if postage in INTERNATIONAL_POSTAGE_TYPES else False
```
Comment (quis) [inline app/notifications/process_notifications.py:150]:
```suggestion
        notification.international = (postage in INTERNATIONAL_POSTAGE_TYPES)
```

File: app/notifications/validators.py [hunk 2]
```diff
@@ -214,3 +215,34 @@ def check_service_letter_contact_id(service_id, letter_contact_id, notification_
             message = 'letter_contact_id {} does not exist in database for service id {}'\
                 .format(letter_contact_id, service_id)
             raise BadRequestError(message=message)
+
+
+def validate_address(service, letter_data):
+    address = PostalAddress.from_personalisation(
+        letter_data,
+        allow_international_letters=(INTERNATIONAL_LETTERS in str(service.permissions)),
```
Comment (quis) [inline app/notifications/validators.py:223]:
```suggestion
        allow_international_letters=service.has_permission(INTERNATIONAL_LETTERS),
```

File: app/notifications/process_notifications.py (hunk 1 above)
Comment (servingUpAces) [reply app/notifications/process_notifications.py:150]: Ah - thanks. I meant to change that

File: app/notifications/validators.py (hunk 2 above)
Comment (servingUpAces) [reply app/notifications/validators.py:223]: service can either be a serialised json object or a db.Model type. If it's serialised there isn't a has_permissions object, if its a db.Model permissions is a resultset. So `(INTERNATIONAL_LETTERS in str(service.permissions))` works for both cases.

Comment (quis) [reply app/notifications/validators.py:223]: Think there’s a better way (add a `has_permission` method to `SerialisedService`) but that’s a bigger refactor so best to leave out of this PR.

File: tests/app/celery/test_letters_pdf_tasks.py [hunk 3]
```diff
@@ -646,6 +646,54 @@ def test_process_sanitised_letter_with_valid_letter(
...
+    template_preview_bucket_name = current_app.config['LETTER_SANITISE_BUCKET_NAME']
+    destination_bucket_name = current_app.config['LETTERS_PDF_BUCKET_NAME']
+    conn = boto3.resource('s3', region_name='eu-west-1')
+    conn.create_bucket(Bucket=scan_bucket_name)
+    conn.create_bucket(Bucket=template_preview_bucket_name)
+    conn.create_bucket(Bucket=destination_bucket_name)
+
+    s3 = boto3.client('s3', region_name='eu-west-1')
+    s3.put_object(Bucket=scan_bucket_name, Key=filename, Body=b'original_pdf_content')
+    s3.put_object(Bucket=template_preview_bucket_name, Key=filename, Body=b'sanitised_pdf_content')
```
Comment (quis) [inline tests/app/celery/test_letters_pdf_tasks.py:673]: Shouldn’t this be mocked? Feels a bit weird to be connecting to an external service that could be unavailable from an integration test.

Comment (servingUpAces) [reply tests/app/celery/test_letters_pdf_tasks.py:673]: That's what @mock_s3 does

Comment (quis) [reply tests/app/celery/test_letters_pdf_tasks.py:673]: Oh, that’s fancy

**Pull request merged**